src/scene_battle_rpg2k3.cpp

```diff
@@ -427,9 +428,33 @@ void Scene_Battle_Rpg2k3::UpdateAnimations() {
 					enemy_cursor->SetX(enemy->GetBattlePosition().x + sprite->GetWidth() / 2 + 2);
 					enemy_cursor->SetY(enemy->GetBattlePosition().y - enemy_cursor->GetHeight() / 2);
 
-					auto* state = enemy->GetSignificantState();
-					if (state) {
-						help_window->SetText(ToString(state->name), state->color);
+					std::vector<lcf::rpg::State*> ordered_states;
+					const auto& states = enemy->GetStates();
+					for (size_t i = 0; i < states.size(); ++i) {
+						if (states[i] > 0) {
+							lcf::rpg::State* state = lcf::ReaderUtil::GetElement(lcf::Data::states, i + 1);
+							ordered_states.push_back(state);
+						}
+					}
+
+					if (ordered_states.size() > 0) {
+						std::sort(ordered_states.begin(), ordered_states.end(), [](lcf::rpg::State* lhs, lcf::rpg::State* rhs) {
+							if (lhs->priority == rhs->priority) {
+								return lhs->ID > rhs->ID;
+							} else {
+								return lhs->priority > rhs->priority;
+							}
+						});
+
+						help_window->Clear();
+						int state_counter = 0;
+						int x = 0;
+						for (lcf::rpg::State* state : ordered_states) {
+							std::string state_name = fmt::format("{:9s}", state->name);
+							help_window->AddText(x, state_name, state->color, Text::AlignLeft, false);
+							x += Font::Default()->GetSize(state_name).width;
```

Comment:
Instead of recomputing the size here, make `AddText()` return the size and reuse it from there.

Reply:
Fixed.